streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -213,6 +212,30 @@ public <K, V> void send(final String topic,
                     keyClass,
                     valueClass),
                 exception);
+        } catch (final SerializationException exception) {
+            final ProducerRecord<K, V> record = new ProducerRecord<>(topic, partition, timestamp, key, value, headers);
+            final ProductionExceptionHandler.ProductionExceptionHandlerResponse response;
+            try {
+                response = productionExceptionHandler.onSerializationException(record, exception);
+            } catch (final Exception e) {
+                log.error("Fatal handling serialization exception on record {}", record, e);
+                recordSendError(topic, e, null);
+                return;
+            }
+
+            if (response == ProductionExceptionHandlerResponse.FAIL) {
+                recordSendError(topic, exception, null);
```

Comment:
Instead of calling `recordSendError` should we just re-throw directly -- otherwise, we might double log and worse actually even call `handle(...)` and it would be very confusing to call the handler (even if two different methods) twice for the same error, and even worse, the handler could not change it's mind and return `CONTINUE` on the second call.